src/ring_theory/adjoin_root.lean

```diff
@@ -124,15 +126,49 @@ by rw [← mk_C x, lift_mk, eval₂_C]
 @[simp] lemma lift_comp_of : (lift i a h).comp (of f) = i :=
 ring_hom.ext $ λ _, @lift_of _ _ _ _ _ _ _ h _
 
+variables (f) [algebra R S]
+
 /-- Produce an algebra homomorphism `adjoin_root f →ₐ[R] S` sending `root f` to
 a root of `f` in `S`. -/
-def alg_hom [algebra R S] (f : polynomial R) (x : S) (hfx : aeval x f = 0) : adjoin_root f →ₐ[R] S :=
-{ commutes' := λ r, show lift _ _ hfx r = _, from lift_of,
-  .. lift (algebra_map R S) x hfx }
+def alg_hom (x : S) (hfx : aeval x f = 0) : adjoin_root f →ₐ[R] S :=
```

Comment:
```suggestion
def alg_hom (x : S) (hfx : aeval x f = 0) : adjoin_root f →ₐ[R] S :=
```
In hindsight, I feel like this definition should have been called `lift`... (Note that it's also defined in terms of `lift`.)

Reply:
this could be renamed to something like alg_lift, if you want

Reply:
I suggest `lift_hom`. That way it will show up in the tab-completion next to `lift` and people will realise that they can choose whatever is best.